server/remote/gcal/call.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package gcal
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/pkg/errors"
+	msgraph "github.com/yaegashi/msgraph.go/v1.0"
+)
+
+func (c *client) CallJSON(method, path string, in, out interface{}) (responseData []byte, err error) {
+	contentType := "application/json"
+	buf := &bytes.Buffer{}
+	err = json.NewEncoder(buf).Encode(in)
+	if err != nil {
+		return nil, err
+	}
+	return c.call(method, path, contentType, buf, out)
+}
+
+func (c *client) CallFormPost(method, path string, in url.Values, out interface{}) (responseData []byte, err error) {
+	contentType := "application/x-www-form-urlencoded"
+	buf := strings.NewReader(in.Encode())
+	return c.call(method, path, contentType, buf, out)
+}
+
+func (c *client) call(method, path, contentType string, inBody io.Reader, out interface{}) (responseData []byte, err error) {
```

Comment:
Probably we will never have this situation here, but do we ever consider an empty body? (i.e. nil)